node_modules/@node-rs/crc32/browser.js Module not found: Can't resolve '@node-rs/crc32-wasm32-wasi'

After cloning the repo and installing the devDeps, I had to manually install the `nw` package as well, because it didn't come included in `package.json`, but is imported in `pages/index.jsx`.

Upon fixing that and running `npm run dev`, it spits the following log:

```bash

> dev
> concurrently -k npm:dev:website npm:dev:desktop

(node:2308) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[dev:desktop] 
[dev:desktop] > dev:desktop
[dev:desktop] > wait-on http-get://localhost:4000 && nwbuild --mode=run --version=0.72.0 --flavor=sdk --no-glob .
[dev:desktop]
[dev:website]
[dev:website] > dev:website
[dev:website] > next dev -p 4000
[dev:website]
[dev:website]   ▲ Next.js 14.2.3
[dev:website]   - Local:        http://localhost:4000
[dev:website]
[dev:website]  ✓ Starting...
[dev:website]  ✓ Ready in 2.1s
[dev:website]  ○ Compiling / ...
[dev:website]  ⨯ ./node_modules/@node-rs/crc32/browser.js:1:1
[dev:website] Module not found: Can't resolve '@node-rs/crc32-wasm32-wasi'
[dev:website]
[dev:website] https://nextjs.org/docs/messages/module-not-found
[dev:website]
[dev:website] Import trace for requested module:
[dev:website] ./node_modules/yauzl-promise/lib/zip.js
[dev:website] ./node_modules/yauzl-promise/lib/index.js
[dev:website] ./node_modules/yauzl-promise/index.js
[dev:website] ./node_modules/nw/src/decompress.js
[dev:website] ./node_modules/nw/src/get.js
[dev:website] ./node_modules/nw/src/index.js
[dev:website] ./pages/index.jsx
```

The *WebAssembly binaries* can only be installed with the `--force` flag on `npm`.

After installing that, another error pops up:

```bash

> dev
> concurrently -k npm:dev:website npm:dev:desktop

(node:9612) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[dev:website]
[dev:website] > dev:website
[dev:website] > next dev -p 4000
[dev:website]
[dev:desktop]
[dev:desktop] > dev:desktop
[dev:desktop] > wait-on http-get://localhost:4000 && nwbuild --mode=run --version=0.72.0 --flavor=sdk --no-glob .
[dev:desktop]
[dev:website] ▲ Next.js 14.2.3
[dev:website] - Local: http://localhost:4000
[dev:website]
[dev:website] ✓ Starting...
[dev:website] ✓ Ready in 2.8s
[dev:website] ○ Compiling / ...
[dev:website] ⨯ ./node_modules/@node-rs/crc32-wasm32-wasi/crc32.wasm32-wasi.wasm?url
[dev:website] Module parse failed: Unexpected character '' (1:0)
[dev:website] The module seem to be a WebAssembly module, but module is not flagged as WebAssembly module for webpack.
[dev:website] BREAKING CHANGE: Since webpack 5 WebAssembly is not enabled by default and flagged as experimental feature.
[dev:website] You need to enable one of the WebAssembly experiments via 'experiments.asyncWebAssembly: true' (based on async modules) or 'experiments.syncWebAssembly: true' (like webpack 4, deprecated).
[dev:website] For files that transpile to WebAssembly, make sure to set the module type in the 'module.rules' section of the config (e. g. 'type: "webassembly/async"').
[dev:website] (Source code omitted for this binary file)
[dev:website]
[dev:website] Import trace for requested module:
[dev:website] ./node_modules/@node-rs/crc32-wasm32-wasi/crc32.wasm32-wasi.wasm?url
[dev:website] ./node_modules/@node-rs/crc32-wasm32-wasi/crc32.wasi-browser.js
[dev:website] ./node_modules/@node-rs/crc32/browser.js
[dev:website] ./node_modules/yauzl-promise/lib/zip.js
[dev:website] ./node_modules/yauzl-promise/lib/index.js
[dev:website] ./node_modules/yauzl-promise/index.js
[dev:website] ./node_modules/nw/src/decompress.js
[dev:website] ./node_modules/nw/src/get.js
[dev:website] ./node_modules/nw/src/index.js
[dev:website] Terminate batch job (Y/N)? Terminate batch job (Y/N)? PS C:\Users\PC\proj\IMPULSE\most updated\nw-react-next-example>
[dev:desktop] npm run dev:desktop exited with code 1
--> Sending SIGTERM to other processes..
[dev:website] npm run dev:website exited with code 1
```

I'm on Windows 10 and the app refuses to run.

--- a/.npmrc
+++ b/.npmrc
@@ -1,0 +1,1 @@
+nw_build_type=sdk

--- a/package.json
+++ b/package.json
@@ -5,23 +5,25 @@
     "main": "http://localhost:4000",
     "node-remote": "http://localhost:4000",
     "scripts": {
-        "postinstall": "nwbuild --mode=get --version=0.72.0 --flavor=sdk",
+        "postinstall": "base-volta-off-of-nwjs",
         "lint": "next lint",
         "dev": "concurrently -k npm:dev:website npm:dev:desktop",
         "dev:website": "next dev -p 4000",
-        "dev:desktop": "wait-on http-get://localhost:4000 && nwbuild --mode=run --version=0.72.0 --flavor=sdk --no-glob .",
+        "dev:desktop": "wait-on http-get://localhost:4000 && nw .",
         "bld": "npm run bld:website && node ./hooks/pre-nw.js && npm run bld:desktop",
         "bld:website": "next build",
-        "bld:desktop": "nwbuild --mode=build --version=0.72.0 --flavor=sdk --no-glob --outDir=./build/desktop .next"
+        "bld:desktop": "nwbuild --mode=build --flavor=sdk --no-glob --outDir=./build/desktop .next"
     },
     "engines": {
         "node": ">= 19.3.0"
     },
     "devDependencies": {
         "@next/eslint-plugin-next": "^15.1.6",
+        "base-volta-off-of-nwjs": "^1.0.5",
         "concurrently": "^9.1.2",
         "eslint": "^9.19.0",
         "next": "^15.1.6",
+        "nw": "^0.95.0",
         "nw-builder": "^4.13.8",
         "react": "^19.0.0",
         "react-dom": "^19.0.0",

--- a/pages/index.jsx
+++ b/pages/index.jsx
@@ -1,15 +1,30 @@
 import React from "react";
 
+function useNW () {
+    const [nw, setNW] = React.useState(null);
+
+    React.useEffect(() => {
+        if (window.require) {
+            setNW(window.nw);
+        }
+    }, []);
+
+    return nw;
+}
+
 function App() {
+
+    const nw = useNW();
+
     return (
         <div>
             NW.js Next.js Example
             <br />
             <ul>
-                <li>NW.js: {nw.require('process').version['nw']}</li>
-                <li>Chromium: {nw.require('process').version['chromium']}</li>
-                <li>Node.js: {nw.require('process').version['node']}</li>
-                <li>V8: {nw.require('process').version['v8']}</li>
+                <li>NW.js: {nw?.require('process')?.version['nw']}</li>
+                <li>Chromium: {nw?.require('process')?.version['chromium']}</li>
+                <li>Node.js: {nw?.require('process')?.version['node']}</li>
+                <li>V8: {nw?.require('process')?.version['v8']}</li>
             </ul>
         </div>
     );

